### PR TITLE
docs: Redirect Meltano Cloud docs to Arch docs

### PR DESCRIPTION
**Pull request by WillDaSilva**
Another PR will remove the Cloud docs, and clean up other outdated mentions of Meltano Cloud.

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -20,3 +20,9 @@
   to = "https://docs.meltano.com/guide/troubleshooting/:splat"
   status = 301
   force = true
+
+[[redirects]]
+  from = "https://docs.meltano.com/cloud/*"
```
Comment (edgarrmondragon) [inline netlify.toml:25]:
```suggestion
  from = "/cloud/*"
```

Comment (edgarrmondragon) [reply netlify.toml:25]: Actually, I'm not sure if we want to redirect in the preview builds 🤔

Comment (edgarrmondragon) [reply netlify.toml:25]: See for example https://deploy-preview-8609--meltano.netlify.app/cloud/connectors/

Comment (WillDaSilva) [reply netlify.toml:25]: We'll be deleting the Cloud docs anyway in https://github.com/meltano/meltano/pull/8610, so I don't think there's any harm in making the redirect apply to preview builds too

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**